food-truck-api/src/main/java/food/truck/api/truck/TruckService.java

```diff
@@ -0,0 +1,64 @@
+package food.truck.api.truck;
+
+import lombok.NonNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TruckService {
+    @Autowired
+    private TruckRepository truckRepository;
+
+    public List<Truck> findTruck(String name) {
+        return truckRepository.findByName(name);
+    }
+
+    public List<Truck> findTruck(Long userId) {
+        return truckRepository.findByUserId(userId);
+    }
+
+    public Optional<Truck> findTryById(Long truckId) {
+        return truckRepository.findById(truckId);
+    }
+
+    public Truck saveTruck(Truck truck) {
+        return truckRepository.save(truck);
+    }
+
+    public Truck createTruck(Long userId, String name) {
+        var t = new Truck();
+        t.setName(name);
+        t.setUserId(userId);
+        return saveTruck(t);
+    }
+
+    public void deleteTruck(Long truckId) {
+        truckRepository
+            .findById(truckId)
+            .ifPresent(truckRepository::delete);
+    }
+
+    public Truck updateTruck(
+        @NonNull
+        Truck truck,
+        Optional<String> name,
+        Optional<byte[]> menu,
+        Optional<String> textMenu,
+        Optional<Long> priceRating,
+        Optional<String> description,
+        Optional<byte[]> schedule,
+        Optional<String> foodCategory
+    ) {
+        name.ifPresent(truck::setName);
```

Comment:
Kind of curious why this is implemented differently than the other sets.

Reply:
I had done it that way so that the request can omit the name of the truck if they are not updating it (the actual name is non-nullable, so it doesn't make sense to otherwise set it to nothing).

Reply:
I think it would be helpful to do it this way for the other ones, too, so it only changes things that the frontend sent a new value for.